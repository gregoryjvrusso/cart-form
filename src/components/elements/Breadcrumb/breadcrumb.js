import React from "react";
import { arrayOf, bool, shape, string } from "prop-types";
import Flex from "../Flex";
import Circle from "../Circle";
import { ArrowRight, TextBread } from "./breadcrumb.styles";
import { arrowBottom, check } from "../../../assets/index";

const Breadcrumb = ({ data }) => (
  <Flex>
    {data.map((option, index) => (
      <Flex key={option.value} justifyContent="center" alignItems="center">
        {option.state && (
          <Circle size="22" color="red" checked>
            <img src={check} alt={option.value} />
          </Circle>
        )}
        {!option.state && (
          <Circle  size="22" color="red">
            <TextBread>{option.value}</TextBread>
          </Circle>
        )}
        <TextBread mr={2} ml={2}>{option.text}</TextBread>
        {index < data.length - 1 && <ArrowRight src={arrowBottom} />}
      </Flex>
    ))}
  </Flex>
);

Breadcrumb.propTypes = {
  data: arrayOf(
    shape({
      value: string,
      text: string,
      state: bool,
    })
  ),
};

Breadcrumb.displayName = "Breadcrumb";

export default Breadcrumb;
