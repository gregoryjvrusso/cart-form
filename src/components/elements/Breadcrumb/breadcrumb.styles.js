import styled from "styled-components";
import { theme } from "styled-tools";
import Text from "../Text";

export const TextBread = styled(Text)`
  color: ${theme("colors.texts.red")};
  font-size: ${theme("fontSizes.1")}px;
`;

export const ArrowRight = styled.img`
  transform: rotate(270deg);
  margin-right: ${theme("spaces.2")}px;
  margin-left: ${theme("spaces.2")}px;
`;
