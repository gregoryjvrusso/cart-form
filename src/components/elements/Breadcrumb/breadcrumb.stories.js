import React from "react";
import { storiesOf } from "@storybook/react";
import { ThemeProvider } from "styled-components";
import theme from "../../../common/theme";

import Breadcrumb from "../Breadcrumb";

const dataMock = [
  [
    {
      value: "1",
      text: "Carrinho",
      state: true,
    },
    {
      value: "2",
      text: "Pagamento",
      state: false,
    },
    {
      value: "3",
      text: "Confirmação",
      state: false,
    },
  ],
  [
    {
      value: "1",
      text: "Carrinho",
      state: true,
    },
    {
      value: "2",
      text: "Pagamento",
      state: true,
    },
    {
      value: "3",
      text: "Confirmação",
      state: false,
    },
  ],
];
storiesOf("Elements/Breadcrumb", module)
  .add("default", () => (
    <ThemeProvider theme={theme}>
      <Breadcrumb data={dataMock[0]} />
    </ThemeProvider>
  ))
  .add("Payment", () => (
    <ThemeProvider theme={theme}>
      <Breadcrumb data={dataMock[1]} />
    </ThemeProvider>
  ));
