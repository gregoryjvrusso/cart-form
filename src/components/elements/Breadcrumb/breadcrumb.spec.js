import React from "react";
import { render } from "@testing-library/react";
import Breadcrumb from "../Breadcrumb";

const dataMock = [
  [
    {
      value: "1",
      text: "Carrinho",
      state: true,
    },
    {
      value: "2",
      text: "Pagamento",
      state: false,
    },
    {
      value: "3",
      text: "Confirmação",
      state: false,
    },
  ],
  [
    {
      value: "1",
      text: "Carrinho",
      state: true,
    },
    {
      value: "2",
      text: "Pagamento",
      state: true,
    },
    {
      value: "3",
      text: "Confirmação",
      state: false,
    },
  ],
];

describe("Breadcrumb component", () => {
  it("deve conseguir renderizar [snapshot]", async () => {
    const { container } = render(<Breadcrumb data={dataMock[0]} />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it("deve conseguir renderizar [snapshot] com 2 links", () => {
    const { container } = render(<Breadcrumb data={dataMock[1]} />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it("deve conseguir renderizar [snapshot] com 1 links", () => {
    const { getByText } = render(<Breadcrumb data={dataMock[0]} />);
    expect(getByText("2")).toBeTruthy();
  });
});
