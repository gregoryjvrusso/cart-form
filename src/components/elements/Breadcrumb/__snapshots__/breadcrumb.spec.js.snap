// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb component deve conseguir renderizar [snapshot] 1`] = `
<div
  class="sc-AxjAm bjcYmf sc-AxirZ gIeogg"
  data-testid="flex"
>
  <div
    class="sc-AxjAm bjcYmf sc-AxirZ lbZIck"
    data-testid="flex"
  >
    <div
      class="sc-AxjAm caeAtN sc-AxirZ iWzAtR sc-AxiKw dFxbwV"
      color="red"
      data-testid="circle"
      size="22"
    >
      <img
        alt="1"
        src="check.svg"
      />
    </div>
    <p
      class="sc-AxhCb hgfAfG sc-AxhUy ccEAFj"
      data-testid="text"
    >
      Carrinho
    </p>
    <img
      class="sc-AxgMl lfVWhc"
      src="arrow-bottom.svg"
    />
  </div>
  <div
    class="sc-AxjAm bjcYmf sc-AxirZ lbZIck"
    data-testid="flex"
  >
    <div
      class="sc-AxjAm caeAtN sc-AxirZ iWzAtR sc-AxiKw cDKsYq"
      color="red"
      data-testid="circle"
      size="22"
    >
      <p
        class="sc-AxhCb fSxDNJ sc-AxhUy ccEAFj"
        data-testid="text"
      >
        2
      </p>
    </div>
    <p
      class="sc-AxhCb hgfAfG sc-AxhUy ccEAFj"
      data-testid="text"
    >
      Pagamento
    </p>
    <img
      class="sc-AxgMl lfVWhc"
      src="arrow-bottom.svg"
    />
  </div>
  <div
    class="sc-AxjAm bjcYmf sc-AxirZ lbZIck"
    data-testid="flex"
  >
    <div
      class="sc-AxjAm caeAtN sc-AxirZ iWzAtR sc-AxiKw cDKsYq"
      color="red"
      data-testid="circle"
      size="22"
    >
      <p
        class="sc-AxhCb fSxDNJ sc-AxhUy ccEAFj"
        data-testid="text"
      >
        3
      </p>
    </div>
    <p
      class="sc-AxhCb hgfAfG sc-AxhUy ccEAFj"
      data-testid="text"
    >
      Confirmação
    </p>
  </div>
</div>
`;

exports[`Breadcrumb component deve conseguir renderizar [snapshot] com 2 links 1`] = `
<div
  class="sc-AxjAm bjcYmf sc-AxirZ gIeogg"
  data-testid="flex"
>
  <div
    class="sc-AxjAm bjcYmf sc-AxirZ lbZIck"
    data-testid="flex"
  >
    <div
      class="sc-AxjAm caeAtN sc-AxirZ iWzAtR sc-AxiKw dFxbwV"
      color="red"
      data-testid="circle"
      size="22"
    >
      <img
        alt="1"
        src="check.svg"
      />
    </div>
    <p
      class="sc-AxhCb hgfAfG sc-AxhUy ccEAFj"
      data-testid="text"
    >
      Carrinho
    </p>
    <img
      class="sc-AxgMl lfVWhc"
      src="arrow-bottom.svg"
    />
  </div>
  <div
    class="sc-AxjAm bjcYmf sc-AxirZ lbZIck"
    data-testid="flex"
  >
    <div
      class="sc-AxjAm caeAtN sc-AxirZ iWzAtR sc-AxiKw dFxbwV"
      color="red"
      data-testid="circle"
      size="22"
    >
      <img
        alt="2"
        src="check.svg"
      />
    </div>
    <p
      class="sc-AxhCb hgfAfG sc-AxhUy ccEAFj"
      data-testid="text"
    >
      Pagamento
    </p>
    <img
      class="sc-AxgMl lfVWhc"
      src="arrow-bottom.svg"
    />
  </div>
  <div
    class="sc-AxjAm bjcYmf sc-AxirZ lbZIck"
    data-testid="flex"
  >
    <div
      class="sc-AxjAm caeAtN sc-AxirZ iWzAtR sc-AxiKw cDKsYq"
      color="red"
      data-testid="circle"
      size="22"
    >
      <p
        class="sc-AxhCb fSxDNJ sc-AxhUy ccEAFj"
        data-testid="text"
      >
        3
      </p>
    </div>
    <p
      class="sc-AxhCb hgfAfG sc-AxhUy ccEAFj"
      data-testid="text"
    >
      Confirmação
    </p>
  </div>
</div>
`;
